Dumps API - Make dump creation an asynchronous task (#139)
meilisearch/specifications: text/0060-tasks-api.md

| field   | type    | description                     |
|---------|---------|---------------------------------|
| uid      | integer | Unique sequential identifier           |

> | indexUid | string | Unique index identifier. This field is defined to `null` for a `dumpCreation`. |

Reviewer: ```suggestion
| indexUid | string | Unique index identifier. This field is `null` when the task type is `dumpCreation`. |
```

Should we say `string or null` in the type column?

Author: Good point will mention that!

| field      | type    | description                     |
|------------|---------|---------------------------------|
| uid        | integer | Unique sequential identifier           |

> | indexUid   | string  | Unique index identifier. This field is defined to `null` for a `dumpCreation`. |

Reviewer: ```suggestion
| indexUid   | string  | Unique index identifier. This field is `null` when the task type is `dumpCreation`. |
```

| distinctAttribute | `distrinctAttribute` payload string |
| displayedAttributes | `displayedAttributes` payload array |

##### dumpCreation

| name    | description  |
| -----   | ------------ |
| dumpUid | The generated uid of the dump |

> Since the creation of a dump is not a task associated with a particular index, it will only be visible on the `GET` - `/tasks` and `GET` - `tasks/:task_uid` endpoints.

Reviewer: ```suggestion
Since the creation of a dump is not a task associated with a particular index, it is only present on the `GET` - `/tasks` and `GET` - `tasks/:task_uid` endpoints.
```

> Fully qualified and summarized task objects related to a dump creation display a `null` `indexUid` field.

Reviewer: This sentence is a bit confusing :s

- Add dedicated task type names modifying a sub-setting. e.g. `SearchableAttributesUpdate`.
- Reconsider existence of `/indexes/:indexUid/tasks/:taskUid` route since it is similar to `/tasks/:taskUid`.
- Add an archived state for old `tasks`.

> - Indicate the `API Key` identity that added a `task`.

Reviewer: ```suggestion
- Add the `API Key` identity that added a `task`.
```
